cmd/srcd/cmd/sql_test.go

```diff
@@ -0,0 +1,209 @@
+	require := s.Require()
+
+	buf, err := s.RunSQL(context.TODO(), "show tables")
+	require.NoError(err)
+
+	require.Contains(buf.String(), showTablesOutput)
+}
+
+func (s *SQLTestSuite) TestValidQueries() {
+	testCases := []string{
+		"show tables",
+		"SHOW TABLES",
+		"show tables;",
+		"SHOW TABLES;",
+		" show tables ; ",
+		" SHOW TABLES ; ",
+		"	show tables	;	",
+		"	SHOW TABLES	;	",
+		"/* comment */ show tables;",
+		`/* multi line
+			comment */
+			show tables;`,
+	}
+
+	for _, query := range testCases {
+		s.T().Run(query, func(t *testing.T) {
+			assert := assert.New(t)
+
+			buf, err := s.RunSQL(context.TODO(), query)
+			assert.NoError(err)
+
+			assert.Contains(buf.String(), showTablesOutput)
+		})
+	}
+}
+
+func (s *SQLTestSuite) TestWrongQuery() {
+	testCases := []struct {
+		query string
+		err   string
+	}{
+		{
+			query: "show",
+			err:   "rpc error: code = Unknown desc = SQL query failed: Error 1105: unknown error: syntax error at position",
+		},
+		{
+			query: "show tables; show tables",
+			err:   "rpc error: code = Unknown desc = SQL query failed: Error 1105: unknown error: syntax error at position",
+		},
+		{
+			query: "select from repositories",
+			err:   "rpc error: code = Unknown desc = SQL query failed: Error 1105: unknown error: syntax error at position",
+		},
+		{
+			query: "select * from nope",
+			err:   "rpc error: code = Unknown desc = SQL query failed: Error 1105: unknown error: table not found: nope",
+		},
+		{
+			query: "insert into repositories values ('myrepo')",
+			err:   "rpc error: code = Unknown desc = SQL query failed: Error 1105: unknown error: table doesn't support INSERT INTO",
+		},
+		{
+			query: "select nope from repositories",
+			err:   `rpc error: code = Unknown desc = SQL query failed: Error 1105: unknown error: column "nope" could not be found in any table in scope`,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.T().Run(tc.query, func(t *testing.T) {
+			assert := assert.New(t)
+
+			buf, err := s.RunSQL(context.TODO(), tc.query)
+			assert.Error(err)
+
+			assert.Contains(buf.String(), tc.err)
+		})
+	}
+}
+
+func (s *SQLTestSuite) TestREPL() {
+	require := s.Require()
+
+	var out, in bytes.Buffer
+
+	command := s.CommandContext(context.TODO(), "sql")
+	command.Stdout = &out
+	command.Stderr = &out
+	command.Stdin = &in
+
+	io.WriteString(&in, "show tables;\n")
+	io.WriteString(&in, "describe table repositories;\n")
+
+	err := command.Start()
+	require.NoError(err)
+
+	err = command.Wait()
+	require.NoError(err)
+
+	expected := `+--------------+
+|    TABLE     |
++--------------+
+| blobs        |
+| commit_blobs |
+| commit_files |
+| commit_trees |
+| commits      |
+| files        |
+| ref_commits  |
+| refs         |
+| remotes      |
+| repositories |
+| tree_entries |
++--------------+
++---------------+------+
+|     NAME      | TYPE |
++---------------+------+
+| repository_id | TEXT |
++---------------+------+`
+
+	require.Contains(out.String(), expected)
```

Comment:
I would also add test writing to stdin after we got results from the previous command. Just to simulate more real use of the repl.

Reply:
Related to the other conversation: https://github.com/src-d/engine/pull/314#discussion_r264658396

Because the command reads stdin and exits, we are not really in interactive mode, and I think we can't test what you describe.